framework/src/utils/PetscSupport.C

```diff
@@ -733,6 +737,63 @@ setSinglePetscOption(const std::string & name, const std::string & value)
     mooseError("Error setting PETSc option.");
 }
 
+void
+colorAdjacencyMatrix(PetscScalar * adjacency_matrix,
+                     unsigned int size,
+                     unsigned int colors,
+                     std::vector<unsigned int> & vertex_colors,
+                     const char * coloring_algorithm)
+{
+  // Mat A will be a dense matrix from the incoming data structure
+  Mat A;
+  MatCreate(MPI_COMM_SELF, &A);
+  MatSetSizes(A, size, size, size, size);
+  MatSetType(A, MATSEQDENSE);
+  // PETSc requires a non-const data array to populate the matrix
+  MatSeqDenseSetPreallocation(A, adjacency_matrix);
+  MatAssemblyBegin(A, MAT_FINAL_ASSEMBLY);
+  MatAssemblyEnd(A, MAT_FINAL_ASSEMBLY);
+
+  // Sparse matrix
+  Mat B;
+  MatConvert(A, MATAIJ, MAT_INITIAL_MATRIX, &B);
```

Comment:
@fdkong - Rather than fill all these entries manually. I think I'll leave the creation of the dense matrix there followed by using a `MatConvert` method with `MAT_INPLACE_MATRIX`. While that still may not be the optimal way to build a matrix. These are very small matrices. It does get rid of the need for building A then B.